x-pack/plugins/actions/server/builtin_action_types/lib/get_node_tls_options.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { PeerCertificate } from 'tls';
+import { TLSSettings } from '../../types';
+
+export function getNodeTLSOptions(
+  verificationMode?: string
+): {
+  rejectUnauthorized?: boolean;
+  checkServerIdentity?: ((host: string, cert: PeerCertificate) => Error | undefined) | undefined;
+} {
+  const agentOptions: {
+    rejectUnauthorized?: boolean;
+    checkServerIdentity?: ((host: string, cert: PeerCertificate) => Error | undefined) | undefined;
+  } = {};
+  if (!!verificationMode) {
+    switch (verificationMode) {
+      case 'none':
+        agentOptions.rejectUnauthorized = false;
+        break;
+      case 'certificate':
+        agentOptions.rejectUnauthorized = true;
+        // by default, NodeJS is checking the server identify
+        agentOptions.checkServerIdentity = () => undefined;
+        break;
+      case 'full':
+        agentOptions.rejectUnauthorized = true;
+        break;
+      default:
```

Comment:
I'm a little worried about the thrown error here - my worry is that we may be using this function in places where we weren't expecting errors to be thrown before.

I'm guessing this isn't a problem in practice - the values are presumably checked at startup, so couldn't actually be anything other than the three valid values; and we would be catching this inside the action executor anyway, which should already handle any kind of random error thrown by an action executor.

It seems like the only other option would be log a warning/error, and use the settings for `full`.

So, I _think_ throwing the error will be fine, but thought I'd mention it in case you think there may be some other case where it could cause problems.

Reply:
I understand your concerns about the throwing an error and I would rather agree that logging a warning message here with the default setting to full should be a safer option for us.